integrations/spark/spark-job-dataquality/spark-job-dataquality-app/src/main/java/com/thinkbiganalytics/spark/dataquality/checker/DataQualityChecker.java

```diff
@@ -20,96 +20,398 @@
-    private String categoryName;
-    private String feedName;
 
-    public static void main(String[] args) {
+    private FlowAttributes flowAttributes;
+    private Map<String, DataQualityRule> availableRules;
+    private List<DataQualityRule> ruleList;
 
+    public static void main(String[] args) {
         log.info("Running DataQualityChecker with these command line args: " + StringUtils.join(args, ","));
 
-        if (args.length < 2) {
-            System.out.println("Expected command line args: <hive-schema-name> <hive-table-name>");
+        if (args.length < 1) {
+            System.out.println("Expected command line args: <path-to-attribute-file>");
             System.exit(1);
         }
 
         try {
             ApplicationContext ctx = new AnnotationConfigApplicationContext("com.thinkbiganalytics.spark");
             DataQualityChecker app = ctx.getBean(DataQualityChecker.class);
-            app.setArguments(args[0], args[1]);
-            app.doDataQualityChecks();
+
+            app.setArguments(args[0]);
+            app.setAvailableRules();
+            app.setActiveRules();
+
+            boolean isSuccess = app.doDataQualityChecks();
+            if (isSuccess) {
+                log.info("DataQualityChecker has PASSED.");
+            } else {
+                log.info("DataQualityChecker has FAILED.");
+                System.exit(1);
+            }
+
+        } catch (MissingRuleException e) {
+            log.error("One or more rules are not available: {}", e.getMessage());
+            System.exit(1);
         } catch (Exception e) {
             log.error("Failed to perform data quality checks: {}", e.getMessage());
             System.exit(1);
         }
+    }
 
-        log.info("DataQualityChecker has finished.");
+    public DataQualityChecker() {
+        flowAttributes = new FlowAttributes();
+        ruleList = new ArrayList<DataQualityRule>();
+        availableRules = new HashMap<>();
     }
 
-    public void setArguments(String categoryName, String feedName) {
-        this.categoryName = categoryName;
-        this.feedName = feedName;
+    /**
+     * Add all the available rules Adds all the data quality rules used by the checker
+     */
+    protected void setAvailableRules() {
+        addAvailableRule(new SourceToFeedCountRuleImpl());
+        addAvailableRule(new RowCountRuleImpl());
+        addAvailableRule(new InvalidRowTotalCountRuleImpl());
+        addAvailableRule(new InvalidRowPercentRuleImpl());
     }
 
-    protected HiveContext getHiveContext() {
-        return hiveContext;
+    /**
+     * Sets the active rules based on the available rule list. The flow attribute dq.active.rules is
+     * used to specify active rules. If this attribute does not exist, all available rules are
+     * used.<br>
+     * If any rule does not exist, an exception is thrown
+     * 
+     * @throws MissingRuleException
+     */
+    protected void setActiveRules() throws MissingRuleException {
```

Comment:
It would be ok to not run any rule if attribute 'dq.active.rules' is absent. DEFAULT_DQ_ACTIVE_RULES_VALUE should be "NONE" (instead of "ALL"). This will make things explicit and clear for the end user. The condition on line 142, and associated unit tests may need to change accordingly.